photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala

```diff
@@ -114,32 +117,37 @@ class Driver(val params: Params, val sparkContext: SparkContext, val logger: Log
     // Load the model from HDFS, ignoring the feature index loader
     val (gameModel, _) =
       ModelProcessingUtils.loadGameModelFromHDFS(Some(featureShardIdToIndexMapLoader), gameModelInputDir, sparkContext)
+    gameModel.persist(StorageLevel.FREQUENT_REUSE_RDD_STORAGE_LEVEL)
 
-    logger.debug(s"Loaded game model summary:\n${gameModel.toSummaryString}")
+    if (logDatasetAndModelStats) {
+      logger.debug(s"Loaded game model summary:\n${gameModel.toSummaryString}")
+    }
 
-    val scores = gameModel.score(gameDataSet).persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL).materialize()
+    val scores = gameModel.score(gameDataSet)
+      .persistRDD(StorageLevel.FREQUENT_REUSE_RDD_STORAGE_LEVEL).materialize()
 
+    gameDataSet.unpersist()
     gameModel.unpersist
     scores
   }
 
   /**
    * Save the computed scores to HDFS with auxiliary info
    *
-   * @param gameDataSet The GAME data set
    * @param scores The computed scores
    */
-  protected def saveScoresToHDFS(
-      gameDataSet: RDD[(Long, GameDatum)],
-      scores: KeyValueScore): Unit = {
+  protected def saveScoresToHDFS(scores: KeyValueScore): Unit = {
 
     // Take the offset information into account when writing the scores to HDFS
```

Comment:
Ideally, we would get rid of the `Driver` classes, or keep them around as examples of how Photon-ML users may want to string together several API calls. In such a case, the code responsible for outputting scored records in AVRO format should be its own function made available through the client to whomever wants to use it and in whatever manner. The Driver shouldn't know or care about how the scores are saved, just that they are (on which note, the method should maybe be renamed to `saveScoredItermsToHdfsAsAvro`). Logical chunks of code responsible for only one action should be encapsulated and made available in a logical location to encourage reusing them.